src/test/java/uk/ac/cam/cl/dtg/isaac/api/managers/GameManagerTest.java

```diff
@@ -92,4 +95,120 @@ public void getNextQuestionsForFilter_appliesExclusionFilterForDeprecatedQuestio
     assertEquals(deprecatedFilter.getOperator(), Constants.BooleanOperator.NOT);
     assertEquals(deprecatedFilter.getValues(), Collections.singletonList("true"));
   }
+
+  @Test
+  public void generateRandomQuestions_returnsCorrectNumberOfQuestions() throws ContentManagerException {
+
+    // Arrange
+    int limit = 5;
+
+    var gameManager = new GameManager(
+        this.dummyContentManager,
+        this.dummyGameboardPersistenceManager,
+        this.dummyMapper,
+        this.dummyQuestionManager,
+        "latest"
+    );
+
+    List<IsaacQuestionPageDTO> questions = new ArrayList<>();
+    for (int i = 0; i < limit; i++) {
+      IsaacQuestionPageDTO question = createMock(IsaacQuestionPageDTO.class);
+      question.setSupersededBy(null);
+      questions.add(question);
+    }
+    var resultsWrapper = new ResultsWrapper(questions, (long) limit);
+
+    // Create a ResultsWrapper with a specific number of questions
+    expect(dummyContentManager.findByFieldNamesRandomOrder(
+        anyObject(),
+        anyInt(),
+        anyInt(),
+        anyLong())
+    ).andReturn(resultsWrapper)
+        .times(1);
+    replay(dummyContentManager);
+
+    // Act
+    var result = gameManager.generateRandomQuestions(new GameFilter(), limit);
+
+    // Assert
+    // Check that the result has the correct number of questions
+    assertEquals(limit, result.size());
+  }
+
+  @Test
+  public void generateRandomQuestions_appliesExclusionFilterForSupersededQuestions() throws ContentManagerException {
+
+    // Arrange
+    int limit = 5;
+    List<IsaacQuestionPageDTO> questions = new ArrayList<>();
+    for (int i = 0; i < limit; i++) {
+      IsaacQuestionPageDTO question = createMock(IsaacQuestionPageDTO.class);
+      questions.add(question);
+    }
+    questions.get(1).setSupersededBy("test_string");
+    var resultsWrapper = new ResultsWrapper(questions, (long) limit);
+
+    // Create a ResultsWrapper with a specific number of questions
+    expect(dummyContentManager.findByFieldNamesRandomOrder(
+        anyObject(),
+        anyInt(),
+        anyInt(),
+        anyLong())
+    ).andReturn(resultsWrapper)
+        .times(2);
+    replay(dummyContentManager);
+
+    var gameManager = new GameManager(
+        this.dummyContentManager,
+        this.dummyGameboardPersistenceManager,
+        this.dummyMapper,
+        this.dummyQuestionManager,
+        "latest"
+    );
+
+    // Act
+    var result = gameManager.generateRandomQuestions(new GameFilter(), limit);
+
+    // Assert
+    // Check that the result has the correct number of questions
+    assertEquals(limit, result.size());
```

Comment:
This is not right. You're testing that the superseded questions are being excluded, so it should have returned 4 results instead of 5 (according to your implementation). I think the reason it's failing is because of line 149, since you are calling a setter on a mock, so that's not going to do anything.